File: StraussModule/core/StraussModel.js
export class StraussModel {
    constructor(theta, sigma) {
        this.theta = theta;
        this.sigma = sigma;
    }

    computeDelta(graph,i,j){

        const edgeExists=graph.hasEdge(i,j);
        const deltaEdges=edgeExists ? -1 : 1;
        const commonNeighbors=graph.commonNeighbors(i,j);
        const deltaTriangles=edgeExists
            ? -commonNeighbors
            : commonNeighbors;

        const result=
            this.theta*deltaEdges+
            this.sigma*deltaTriangles;

        console.log(
            "edgeExists=",edgeExists,
            "deltaEdges=",deltaEdges,
            "commonNeighbors=",commonNeighbors,
            "deltaTriangles=",deltaTriangles,
            "theta=",this.theta,
            "sigma=",this.sigma,
            "result=",result
        );

        return result;
    }
    calculateEnergy(graph){
        const edges = graph.countEdges();
        const triangles = graph.countTriangles();
        return (this.theta * edges + this.sigma * triangles);
    }
}

File: StraussModule/python_core/StraussModel.py
class StraussModel:

########################################################################################################
    def __init__(self, theta, sigma):
        self.theta = theta
        self.sigma = sigma

########################################################################################################
    def computeDelta(self, graph, i, j):

        edgeExists = graph.hasEdge(i, j)
        deltaEdges = -1 if edgeExists else 1
        commonNeighbors = graph.commonNeighbors(i, j)
        deltaTriangles = (-commonNeighbors if edgeExists else commonNeighbors)
        result = (self.theta * deltaEdges + self.sigma * deltaTriangles)

        return result

########################################################################################################
    def calculateEnergy(self, graph):

        edges = graph.countEdges()
        triangles = graph.countTriangles()
        return (self.theta * edges + self.sigma * triangles)